fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RelOpValue.java

```diff
@@ -149,15 +150,19 @@ public Optional<QueryPredicate> toQueryPredicate(@Nonnull final CorrelationIdent
                 // it seems this is a constant expression, try to evaluate it.
                 return tryBoxSelfAsConstantPredicate();
             }
-        } else if (childrenCount == 2) {
+        }
+        Verify.verify(childrenCount == 2);
```

Comment:
The contract of `toQueryPredicate` is best-effort to transform `this` value into a `QueryPredicate`, this is reflected by returning an _Optional_ object, so we can propagate failure to find a mapping by returning `Optional.empty()`.

Theoretically, a component on top, be it a plan generator, could choose to generate a different `Value`, however now that we add a `Verify` the contract has been altered.

Therefore, I suggest to return `Optional.empty()` if `childrenCount != 2`.

Reply:
done